### ISSUE TITLE
Plugins not working

### ISSUE THREAD

**Issue by alanyinjs** [AtomLinter/linter-eslint#1189]
## Issue Type
Question

## Issue Description
I have an old .eslintrc.json file from one of my old projects. I would like to use the same settings in a new project. However when I copied it over, atom started complaining about plugins not being found. Those plugins are `eslint-plugin-mocha`, `eslint-plugin-react-native`, `eslint-plugin-react` and `eslint-plugin-flowtype`.

They seem to work fine in my old project. I have got them installed under `.atom/packages/linter-eslint/eslint-plugin-xxx(whatever the name is)`.

Atom is complaining by highlighting the first line of code of whatever `.js` file I open and displaying a message saying the following:
<img width="1086" alt="screen shot 2018-10-26 at 2 39 19 pm" src="https://user-images.githubusercontent.com/25688258/47543246-f6f77100-d92c-11e8-9779-872ae2f8d098.png">

No error in the console.

I am also sure that those plugins and linter-eslint itself are installed with required dependencies. I have ran into this issue previously where linter-eslint wasn't working properly because one of the dependencies wasn't installed. 

## Bug Checklist
- [x] Restart Atom
- [x] Verify the `eslint` CLI gives the proper result, while `linter-eslint` does not
- [x] Paste the output of the `Linter Eslint: Debug` command from the Command Palette below

```
Atom version: 1.31.2
linter-eslint version: 8.4.1
ESLint version: 4.15.0
Hours since last Atom restart: 0.1
Platform: darwin
Using bundled fallback ESLint from: /Users/alanyin/.atom/packages/linter-eslint/node_modules/eslint
Current file's scopes: [
  "source.json",
  "meta.structure.dictionary.json",
  "meta.structure.dictionary.value.json",
  "meta.structure.array.json",
  "string.quoted.double.json",
  "punctuation.definition.string.end.json"
]
linter-eslint configuration: {
  "lintHtmlFiles": false,
  "useGlobalEslint": false,
  "showRuleIdInMessage": true,
  "disableWhenNoEslintConfig": true,
  "eslintrcPath": "",
  "globalNodePath": "",
  "advancedLocalNodeModules": "",
  "eslintRulesDirs": [],
  "disableEslintIgnore": false,
  "disableFSCache": false,
  "fixOnSave": false,
  "scopes": [
    "source.js",
    "source.jsx",
    "source.js.jsx",
    "source.babel",
    "source.js-semantic"
  ],
  "rulesToSilenceWhileTyping": [],
  "rulesToDisableWhileFixing": [],
  "ignoreFixableRulesWhileTyping": false
}
```

**Comment by Arcanemagus** [comment]
> Using bundled fallback ESLint from: /Users/alanyin/.atom/packages/linter-eslint/node_modules/eslint

It looks like you haven't installed ESLint locally in that project yet. You'll need to copy the `devDependencies` related to linting from the other project as well (one of the reasons I prefer putting the ESLint configuration in `package.json`).

> They seem to work fine in my old project. I have got them installed under .atom/packages/linter-eslint/eslint-plugin-xxx(whatever the name is).

That is very much _not_ recommended.

I'm marking this as closed since this isn't a bug in `linter-eslint`, however feel free to comment further if you have more questions.

Thanks for taking the time to file an issue!

**Comment by alanyinjs** [comment]
@Arcanemagus 

Hi Landon,

Thanks a lot for the prompt reply. I figured out what the problem was. It seems that the problem is due to clash of local and global packages (I mean, packages in the `.atom/packages/linter-eslint` directory). I tried the same approach this time with a single `.js` file and it seems to work without local installation. I forgot to mention the problem I encountered was setting up with `create-react-app`, which has some default linting setup behind the scenes. 

I do agree with you falling back on `.atom/packages/linter-eslint/` is not a good practice (just the way my company's code base works), as plugins should be project-specific and there were reported clashes between local and global packages (from some previous issues). Local setup does make a lot more sense.

Thanks again for your reply! Very much appreciated.

Kind regards,
Alan
